src/System.Management.Automation/engine/MshMemberInfo.cs

```diff
@@ -4203,9 +4218,9 @@ internal override ReadOnlyPSMemberInfoCollection<T> Match(string name, PSMemberT
         private PSMemberInfoInternalCollection<T> GetInternalMembers(MshMemberMatchOptions matchOptions)
         {
             PSMemberInfoInternalCollection<T> returnValue = new PSMemberInfoInternalCollection<T>();
-            lock (_members)
+            lock (Members)
```

Comment:
Same here, shall we also check for `_members == null` for a shortcut path?

Reply:
Fixed.